applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -20,148 +20,84 @@
-        mEstimateInterpError = default_parameters["hessian_strategy_parameters"]["estimate_interpolation_error"].GetBool();
-        mInterpError = default_parameters["hessian_strategy_parameters"]["interpolation_error"].GetDouble();
-        mMeshConstant = default_parameters["hessian_strategy_parameters"]["mesh_dependent_constant"].GetDouble();
-        mRatioReferenceVariable = default_parameters["anisotropy_parameters"]["reference_variable_name"].GetString();
-        mAnisotropicRatio = default_parameters["anisotropy_parameters"]["hmin_over_hmax_anisotropic_ratio"].GetDouble();
-        mBoundLayer = default_parameters["anisotropy_parameters"]["boundary_layer_max_distance"].GetDouble();
-        mInterpolation = ConvertInter(default_parameters["anisotropy_parameters"]["interpolation"].GetString());
-    } else {
-        mEstimateInterpError = ThisParameters["hessian_strategy_parameters"]["estimate_interpolation_error"].GetBool();
-        mInterpError = ThisParameters["hessian_strategy_parameters"]["interpolation_error"].GetDouble();
-        mMeshConstant = ThisParameters["hessian_strategy_parameters"]["mesh_dependent_constant"].GetDouble();
-        mRatioReferenceVariable = ThisParameters["anisotropy_parameters"]["reference_variable_name"].GetString();
-        mAnisotropicRatio = ThisParameters["anisotropy_parameters"]["hmin_over_hmax_anisotropic_ratio"].GetDouble();
-        mBoundLayer = ThisParameters["anisotropy_parameters"]["boundary_layer_max_distance"].GetDouble();
-        mInterpolation = ConvertInter(ThisParameters["anisotropy_parameters"]["interpolation"].GetString());
-    }
+ComputeHessianSolMetricProcess::ComputeHessianSolMetricProcess(
+    ModelPart& rThisModelPart,
+    ComponentType& rVariable,
+    Parameters ThisParameters
+    ):mThisModelPart(rThisModelPart)
+{
+    // We push the components list
+    mrOriginVariableComponentsList.push_back(rVariable);
+
+    // We check the parameters
+    Parameters default_parameters = GetDefaultParameters();
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+    InitializeVariables(ThisParameters);
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
-template<SizeType TDim, class TVarType>
-void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
+void ComputeHessianSolMetricProcess::Execute()
 {
-    // Iterate in the nodes
-    NodesArrayType& nodes_array = mThisModelPart.Nodes();
-    const int num_nodes = nodes_array.end() - nodes_array.begin();
-
     CalculateAuxiliarHessian();
 
     // Some checks
-    VariableUtils().CheckVariableExists(mVariable, nodes_array);
+    NodesArrayType& nodes_array = mThisModelPart.Nodes();
+    if (mrOriginVariableDoubleList.size() > 0) {
+        VariableUtils().CheckVariableExists(mrOriginVariableDoubleList[0], nodes_array);
```

Comment:
side comment, this function does not take its args as `const`!

Reply:
CheckVariableExists?

Reply:
yes